docs: add page for treemap visualizations (#77185)
metabase/metabase: docs/questions/visualizations/treemap.md

---
title: Treemaps
---

# Treemaps

> Treemap visualizations display hierarchical data as nested rectangles. The value of a metric determines the size of each rectangle, called a leaf. Larger values take up more area than smaller values.

Reviewer: ```suggestion
Treemap visualizations display hierarchical data as nested rectangles. These hierarchies are called trees, and the nodes at the bottom of the hierarchies leaves. The value of a metric determines the size of each leaf. Larger values take up more area than smaller values.
```

Reviewer: It's called a treemap, but I don't think people immediately understand what trees are in this context. There's probably a better way to phrase this, but it's a start.

When you group by two dimensions, the first dimension becomes the parent rectangles. The second dimension nests inside the parent rectangles as leaves.

![Treemap](../images/treemap.png)

## When to use a treemap

Use a treemap to compare how a total breaks down by one categorical dimension, or by a second categorical dimension nested within the first.

For example, you can show order counts by product category, then break each category down by vendor.

## Data shape for a treemap

To create a treemap, create a question that returns a single metric grouped by one or two dimensions.

> When you group by one dimension, a treemap shows one leaf per value. Here's the data shape for a treemap of order count by category:

Reviewer: It's always one leaf per value, but when you have two dimensions you get branches before you reach the leaves (and there are more leaves).

| Category  | Count |
| --------- | ----- |
| Widget    | 414   |
| Gadget    | 350   |
| Doohickey | 334   |
| Gizmo     | 289   |

> When you group by two dimensions, the treemap nests the second dimension inside the first. The first dimension determines the outer rectangles and their colors. The second dimension determines the leaves nested inside the outer rectangles. The value of the metric determines each leaf's size.

Reviewer: I actually wonder if it would help to draw a tree as a node-link diagram to help explain this. Treemaps are a very weird way of looking at hierarchies (but good for comparing size), node-link diagrams are much easier to understand.

Reviewer: These drawings are… a few years old, but to give you a sense of what I mean: https://eagereyes.org/blog/2008/treemaps

- **Grouping:** The dimension that determines the parent categories. To set a custom color for a parent category, click the color swatch next to the category's name. To rename a parent category, click the **...** next to the category's name.
- **Sub-grouping:** The dimension that nests inside each parent category as leaves.
- **Value:** The metric that determines the size of each leaf. To format the value, click the **...** to open the formatting menu. You can set the number style, separator style, number of decimal places, a multiplier, and a prefix or suffix.

### Display

In chart settings, click the **Display** tab to edit how the chart looks.

Use the following toggles to show or hide labels and values:

- **Show parent labels:** Display the name of each parent category.
- **Show parent values:** Display the value for each parent category.
- **Show leaf labels:** Display the name of each leaf.
- **Show leaf values:** Display the value for each leaf.

## Limitations and alternatives

- Treemaps display a single metric grouped by one or two dimensions. If your query returns more than one metric, the treemap uses the metric you select in the **Value** slot.

> - When one leaf is far larger than the other leaves, the small leaves can become too small to see or select. If your values span a wide range, consider a [bar chart](./line-bar-and-area-charts.md) instead.

Reviewer: If the difference is really that large, a bar chart won't show much either though :) But this is a valid point.

In addition, if the comparison is mainly between the category values, and not the values and the whole, a bar chart is the better choice.

> - To show a single level of part-to-whole composition, use a [pie chart](./pie-or-donut-chart.md) instead.

Reviewer: ```suggestion
- To show a single level of part-to-whole comparison, when the number of categories is small, use a [pie chart](./pie-or-donut-chart.md) instead.
```

---
title: Treemaps

> ---

Reviewer 2: Can we get a summary as well?

# Treemaps

Treemap visualizations display hierarchical data as nested rectangles. The value of a metric determines the size of each rectangle, called a leaf. Larger values take up more area than smaller values.

> When you group by two dimensions, the first dimension becomes the parent rectangles. The second dimension nests inside the parent rectangles as leaves.

Reviewer 2: ```suggestion
Treemap visualizations display hierarchical data as nested rectangles. The value of a metric determines the size of each rectangle, called a leaf. Larger values take up more area than smaller values.

When you group a metric by two dimensions, the first dimension becomes the parent rectangles. The second dimension nests inside the parent rectangles as leaves.
```

> When you group by two dimensions, the first dimension becomes the parent rectangles. The second dimension nests inside the parent rectangles as leaves.

Reviewer 2: ```suggestion
When you group by two dimensions, the first dimension becomes the parent categories. The second dimension nests inside the parent rectangles as leaves.
```
The app uses "Parent categories" and "Leaves", so we should stick to those terms.

Reviewer: Should we say, "the first dimension defines the parent categories" instead of "becomes"? I feel like the singular and plural don't agree here.

> ![Treemap](../images/treemap.png)

Reviewer 2: We should probably give some commentary on what we're looking at here:

```suggestion
Here we've grouped the metric, count of products, by two dimensions: first by product category (the larger rectangles), then by vendor (the nested leaves in each product category).

![Treemap](../images/treemap.png)
```

## When to use a treemap

> Use a treemap to compare how a total breaks down by one categorical dimension, or by a second categorical dimension nested within the first.

Reviewer 2: Can we expand on this? Be more specific about when to use a treemap? For example, to compare how a total breaks down by one categorical dimension, you'll often prefer a row/bar chart. Or a donut/pie chart if there are only 2-3 categories. When, specifically, should the treemap prevail?

Reviewer: Agreed. Treemaps are best when you have many items to break down into (where pie charts become unwieldy), and the intended comparison is more part-to-whole than between parts.

For example, you can show order counts by product category, then break each category down by vendor.

## Data shape for a treemap

To create a treemap, create a question that returns a single metric grouped by one or two dimensions.

When you group by one dimension, a treemap shows one leaf per value. Here's the data shape for a treemap of order count by category:

> | Category  | Count |

Reviewer 2: Can we show images of each stage so people can see how Metabase will display this info?

| --------- | ----- |
| Widget    | 414   |
| Gadget    | 350   |
| Doohickey | 334   |
| Gizmo     | 289   |

When you group by two dimensions, the treemap nests the second dimension inside the first. The first dimension determines the outer rectangles and their colors. The second dimension determines the leaves nested inside the outer rectangles. The value of the metric determines each leaf's size.

Here's the data shape for a treemap of order count by category and vendor:

> | Category  | Vendor                     | Count |

Reviewer 2: Image for this table rendered as a treemap as well.

| --------- | -------------------------- | ----- |
| Widget    | Jerrell Gulgowski Inc      | 26    |
| Widget    | Brittany Mueller Inc       | 25    |
| Gadget    | Jerrod McLaughlin LLC      | 24    |
| Gadget    | Miles Ryan Group           | 22    |
| Gizmo     | Mr. Tanya Stracke and Sons | 30    |
| Doohickey | Kuhn-O'Reilly              | 27    |

## Build a query for a treemap

To build a nested treemap:

1. In the query builder, click **Summarize**.
2. Set a metric, such as **Count of rows**.
3. Group by a first dimension, such as **Category**.
4. Add a second grouping, such as **Vendor**.
5. In the visualization picker, select **Treemap**.

> The order of the groupings matters. The first grouping determines the outer rectangles, and the second grouping nests inside.

Reviewer 2: ```suggestion
The order of the groupings matters. The first grouping determines the outer rectangles, and the second grouping nests the leaves in each outer rectangle.
```

## Treemap settings

To open chart settings, click the **Gear** icon in the bottom left of the visualization.

### Data

In chart settings, click the **Data** tab to configure the chart's groupings and value.

- **Grouping:** The dimension that determines the parent categories. To set a custom color for a parent category, click the color swatch next to the category's name. To rename a parent category, click the **...** next to the category's name.

> - **Sub-grouping:** The dimension that nests inside each parent category as leaves.

Reviewer 2: ```suggestion
- **Sub-grouping:** The dimension that nests inside each parent category as leaves. Requires a query with at least two groupings.
```

- **Value:** The metric that determines the size of each leaf. To format the value, click the **...** to open the formatting menu. You can set the number style, separator style, number of decimal places, a multiplier, and a prefix or suffix.

### Display

In chart settings, click the **Display** tab to edit how the chart looks.

Use the following toggles to show or hide labels and values:

- **Show parent labels:** Display the name of each parent category.
- **Show parent values:** Display the value for each parent category.
- **Show leaf labels:** Display the name of each leaf.

> - **Show leaf values:** Display the value for each leaf.

Reviewer 2: ```suggestion
In chart settings, click the **Display** tab to show or hide labels and values for parent categories and leaves.

```

> ## Limitations and alternatives

Reviewer 2: Some of the guidance here is probably better placed in the "when to use treemaps" section.

> - Treemaps display a single metric grouped by one or two dimensions. If your query returns more than one metric, the treemap uses the metric you select in the **Value** slot.

Reviewer 2: Maybe a note about what happens when a query groups by more than two dimensions.

Treemaps are a weaker choice for comparing categories against each other, because area is harder to judge than bar length. For a single dimension with only a few categories, use a [pie chart](./pie-or-donut-chart.md) or a [bar chart](./line-bar-and-area-charts.md).

## Data shape for a treemap

To create a treemap, create a question that returns a single metric grouped by one or two dimensions.

When you group a metric by one dimension, the treemap shows a flat set of leaves with no parent categories. Here's the data shape for a treemap of order count by category:

| Category  | Count |
| --------- | ----- |
| Widget    | 612   |
| Gadget    | 562   |
| Gizmo     | 491   |
| Doohickey | 462   |

![Treemap grouped by one dimension](../images/treemap-one-dimension.png)

When you group a metric by two dimensions, the treemap nests the second dimension inside the first. The first dimension determines the outer rectangles and their colors. The second dimension determines the leaves nested inside the outer rectangles. The value of the metric determines each leaf's size.

> Here's the data shape for a treemap of order count by category and vendor:

Reviewer 2: ```suggestion
Here's the data shape for a treemap of order count by category (the parent category) and vendor (the leaves):
```

| Category  | Vendor                     | Count |
| --------- | -------------------------- | ----- |
| Widget    | Jerrell Gulgowski Inc      | 26    |
| Widget    | Brittany Mueller Inc       | 25    |
| Gadget    | Jerrod McLaughlin LLC      | 24    |
| Gadget    | Miles Ryan Group           | 22    |
| Gizmo     | Mr. Tanya Stracke and Sons | 30    |
| Doohickey | Kuhn-O'Reilly              | 27    |

![Treemap grouped by two dimensions](../images/treemap.png)

## Build a query for a treemap

To build a nested treemap:

1. In the query builder, click **Summarize**.

> 2. Set a metric, such as **Count of rows**.

Reviewer 2: ```suggestion
2. Select a metric, such as **Count of rows**.
```

3. Group by a first dimension, such as **Category**.
4. Add a second grouping, such as **Vendor**.
5. In the visualization picker, select **Treemap**.

The order of the groupings matters. The first grouping determines the outer rectangles, and the second grouping determines the leaves nested in each outer rectangle.

Click a parent category to zoom into that category.

## Drill-through

- **Parent category**: Click a parent category to zoom into that category. To return to the full treemap, click the back arrow.
- **Leaf**: Click a leaf to view the underlying records, break out by another dimension, view automatic insights, or filter by its value.

> Drill-through options are limited for questions written in SQL. For more information, see [drill-through](./drill-through.md).

Reviewer 2: ```suggestion
[Drill-through(./drill-through.md) options are limited for questions written in SQL.
```
